src/transport_ofi.c

```diff
@@ -99,28 +99,124 @@ static inline void init_dt_size(void)
   SHMEM_Dtsize[FI_LONG_DOUBLE_COMPLEX] = sizeof(long double complex);
 }
 
-static int SHM_DT_INT[]=
+static char * SHMEM_DtName[FI_DATATYPE_LAST]=
 {
-  SHM_INTERNAL_SHORT, SHM_INTERNAL_INT, SHM_INTERNAL_LONG,
+  "int8", "uint8", "int16", "uint16", "int32", "uint32", "int64", "uint64"
+  "float", "double", "float complex", "double complex", "long double",
+  "long double complex"
```

Comment:
This is true for how we use the constants in building Dtsize, but with
Dtname you are assuming a specific value.
On Apr 29, 2016 6:05 PM, "Kayla Seager" <notifications@github.com> wrote:

> In src/transport_ofi.c
> <https://github.com/Sandia-OpenSHMEM/SOS/pull/169#discussion_r61651066>:
>
> >  {
> > -  SHM_INTERNAL_SHORT, SHM_INTERNAL_INT, SHM_INTERNAL_LONG,
> > +  "int8", "uint8", "int16", "uint16", "int32", "uint32", "int64", "uint64"
> > +  "float", "double", "float complex", "double complex", "long double",
> > +  "long double complex"
>
> so I checked up on this, these assumptions are encoded in the version
> number you get when you call getinfo and its part of the ABI compatibility.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/Sandia-OpenSHMEM/SOS/pull/169/files/bbc5169dea7ecdb443670bf5e08230172859af7e#r61651066>
>